src/app_add_cycle.rs

```diff
@@ -0,0 +1,308 @@
+
+        // When declaring in a cycle
+        let result = app.add_cycle(vec![rx_a1, rx_a2]);
+
+        // Then observe an error
+        let errors = result.expect_err("expected conflict");
+
+        assert_eq!(
+            errors.len(),
+            1,
+            "Expected 1 error, but found: {:#?}",
+            errors
+        );
+        let one_err = errors.first().unwrap();
+        if let CycleCheckError::UpdatePackResetInMultipleWorkloads { update_pack, .. } = one_err {
+            assert_eq!(*update_pack, type_name::<A>());
+        } else {
+            panic!(
+                "Expected error to be UpdatePackResetInMultipleWorkloads, but found {:#?}",
+                one_err
+            );
+        }
+    }
+
+    #[test]
+    fn test_duplicates_update_packed_ok() {
+        let mut app = setup_app();
+
+        // Given added workload 1 depends on [A] being update packed
+        let rx_a_1 = app.add_plugin_workload_with_info(RxADup);
+        // And adding the SAME workload 2
+        let rx_a_2 = app.add_plugin_workload_with_info(RxADup);
+
+        // When declaring in a cycle
+        let result = app.add_cycle(vec![rx_a_1, rx_a_2]);
+
+        // Then observe ok
+        let errors = result.expect("expected no conflict");
+    }
+
+    #[test]
+    fn test_conflicting_tracked() {
+        let mut app = setup_app();
+
+        // Given added workload 1 depends on [A] being a tracked unique
+        let rx_a1 = app.add_plugin_workload_with_info(RxTrackA1);
+        // And adding workload 2 depends on SAME tracked unique [A]
+        let rx_a2 = app.add_plugin_workload_with_info(RxTrackA2);
+
+        // When declaring in a cycle
+        let result = app.add_cycle(vec![rx_a1, rx_a2]);
+
+        // Then observe an error
+        let errors = result.expect_err("expected conflict");
+
+        assert_eq!(
+            errors.len(),
+            1,
+            "Expected 1 error, but found: {:#?}",
+            errors
+        );
+        let one_err = errors.first().unwrap();
+        if let CycleCheckError::TrackedUniqueResetInMultipleWorkloads { tracked_unique, .. } =
+            one_err
+        {
+            assert_eq!(*tracked_unique, type_name::<A>());
+        } else {
+            panic!(
+                "Expected error to be TrackedUniqueResetInMultipleWorkloads, but found {:#?}",
+                one_err
+            );
+        }
+    }
+
+    #[test]
+    fn test_duplicates_tracked_ok() {
+        let mut app = setup_app();
+
+        // Given added workload 1 depends on [A] being tracked unique
+        let rx_a_1 = app.add_plugin_workload_with_info(RxTrackADup);
+        // And adding the SAME workload 2
+        let rx_a_2 = app.add_plugin_workload_with_info(RxTrackADup);
+
+        // When declaring in a cycle
+        let result = app.add_cycle(vec![rx_a_1, rx_a_2]);
+
+        // Then observe ok
+        let errors = result.expect("expected no conflict");
+    }
+
+    impl crate::Plugin for RxA1 {
```

Comment:
All of those impls could probably be done with a macro for greatly reducing the code, but I'm not sure it's worth the hassle.

Reply:
I'll keep this in mind for future!